inc/sitemaps/class-sitemaps.php

```diff
@@ -0,0 +1,429 @@
+		}
+	}
+
+	/**
+	 * Register your own XSL file. Call this during 'init'.
+	 *
+	 * @param string   $name     The name of the XSL file.
+	 * @param callback $function Function to build your XSL file.
+	 * @param string   $rewrite  Optional. Regular expression to match your sitemap with.
+	 */
+	public function register_xsl( $name, $function, $rewrite = '' ) {
+		add_action( 'wpseo_xsl_' . $name, $function );
+		if ( ! empty( $rewrite ) ) {
+			add_rewrite_rule( $rewrite, 'index.php?xsl=' . $name, 'top' );
+		}
+	}
+
+	/**
+	 * Set the sitemap current page to allow creating partial sitemaps with wp-cli
+	 * in an one-off process.
+	 *
+	 * @param integer $current_page The part that should be generated.
+	 */
+	public function set_n( $current_page ) {
+		if ( is_scalar( $current_page ) && intval( $current_page ) > 0 ) {
+			$this->current_page = intval( $current_page );
+		}
+	}
+
+	/**
+	 * Set the sitemap content to display after you have generated it.
+	 *
+	 * @param string $sitemap The generated sitemap to output.
+	 */
+	public function set_sitemap( $sitemap ) {
+		$this->sitemap = $sitemap;
+	}
+
+	/**
+	 * Set a custom stylesheet for this sitemap. Set to empty to just remove the default stylesheet.
+	 *
+	 * @deprecated
+	 *
+	 * @param string $stylesheet Full xml-stylesheet declaration.
+	 */
+	public function set_stylesheet( $stylesheet ) {
+		$this->renderer->set_stylesheet( $stylesheet );
+	}
+
+	/**
+	 * Set as true to make the request 404. Used stop the display of empty sitemaps or invalid requests.
+	 *
+	 * @param bool $bool Is this a bad request. True or false.
+	 */
+	public function set_bad_sitemap( $bool ) {
+		$this->bad_sitemap = (bool) $bool;
+	}
+
+	/**
+	 * Prevent stupid plugins from running shutdown scripts when we're obviously not outputting HTML.
+	 *
+	 * @since 1.4.16
+	 */
+	public function sitemap_close() {
+		remove_all_actions( 'wp_footer' );
+		die();
+	}
+
+	/**
+	 * Hijack requests for potential sitemaps and XSL files.
+	 *
+	 * @param \WP_Query $query Main query instance.
+	 */
+	public function redirect( $query ) {
+
+		if ( ! $query->is_main_query() ) {
+			return;
+		}
+
+		$xsl = get_query_var( 'xsl' );
+
+		if ( ! empty( $xsl ) ) {
+			$this->xsl_output( $xsl );
+			$this->sitemap_close();
+
+			return;
+		}
+
+		$type = get_query_var( 'sitemap' );
+
+		if ( empty( $type ) ) {
+			return;
+		}
+
+		$this->set_n( get_query_var( 'sitemap_n' ) );
+
+		/**
+		 * Filter: 'wpseo_enable_xml_sitemap_transient_caching' - Allow disabling the transient cache
+		 *
+		 * @api bool $unsigned Enable cache or not, defaults to true
+		 */
+		$caching = apply_filters( 'wpseo_enable_xml_sitemap_transient_caching', true );
+
+		if ( $caching ) {
+			do_action( 'wpseo_sitemap_stylesheet_cache_' . $type, $this );
+			$this->sitemap   = get_transient( 'wpseo_sitemap_cache_' . $type . '_' . $this->current_page );
+			$this->transient = ! empty( $this->sitemap );
+		}
+
+		if ( empty( $this->sitemap ) ) {
```

Comment:
If you keep this logic intact, it would make much more sense to me to check if `this->transient === false`.